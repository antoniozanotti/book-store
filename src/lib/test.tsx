import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
