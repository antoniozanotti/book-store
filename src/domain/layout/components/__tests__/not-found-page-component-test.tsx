import { render, screen } from "@/lib/test";
import { NotFoundPageComponent } from "../not-found-page-component";

describe("NotFoundPageComponent", () => {
  it("should render", async () => {
    const { container } = render(<NotFoundPageComponent />);
    expect(container).toMatchSnapshot();
  });
});