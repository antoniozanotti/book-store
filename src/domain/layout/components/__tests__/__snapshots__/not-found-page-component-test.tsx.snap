// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPageComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <main
      class="p-10"
    >
      <h1
        class="font-primary text-pretty text-5xl sm:text-6xl md:text-7xl lg:text-8xl"
      >
        Not Found
      </h1>
      <p>
        Could not find requested resource.
      </p>
      <a
        href="/"
      >
        Return Home.
      </a>
    </main>
  </div>
</div>
`;
